### PR TITLE
Add release notes for v1.2.0

### PR DESCRIPTION
**Pull request by lunny**
As titile.

### CODE REVIEW

**Comment by lafriks** [comment]
Shouldn't RC changelog be removed as these are duplicate with 1.2.0 final?

**Comment by lunny** [comment]
@lafriks I have no idea about that. @tboerger @bkcsoft @appleboy ?

**Comment by strk** [comment]
+1 for removing RC section and I'd also get rid of the BUGS section as all bug fixes for existing features should go to the stable branch (1.1). This should be only new features and (hopefully none) breaking changes. Please see also #2608 because there's a lot of redundancy in this CHANGELOG

**Comment by bkcsoft** [comment]
+1 for removing RC sections

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,494 @@
...
+  * Per issue/PR watch/unwatch (#1410)
+  * Allow ENABLE_OPENID_SIGNUP to depend on DISABLE_REGISTRATION (#1369)
+  * Fix FCGI (over TCP) support (#1368)
+  * Repo size in admin panel (#1482)
+  * Show user OpenID URIs in their profile (#1314)
+  * Add change-password admin command (#1304)
+  * feat: Only use issue and wiki on repo. (#1297)
+  * Allow push to init a wiki repo (#1279)
+  * Dockerfile for aarch64 (#1128) (#1130)
+* SECURITY
```
Comment (tboerger) [inline CHANGELOG.md]: Security and breaking changes should be on the top of the list

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,494 @@
...
+  * Change two factor code entry box from text to number (#1733)
+  * gofmt (#1710)
+  * Hide the create organization button (in dashboard/organization section) (#1705)
+  * Correct spelling mistakes (#1703)
+  * Adds Parent property to the repo API (#1687)
+  * Add configuration option for default permission to create Organizations (#1686)
+  * Remove sha1 hash display in repository table (#1678)
+  * Download files to their original filename (#1676)
+  * Exposes in API the Repo entity's Size and IsBare property
+ (#1668)
```
Comment (tboerger) [inline CHANGELOG.md]: This new line should be removed

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,494 @@
...
+  * Only run coverage on merges/pushes to master (#1783)
+  * Remove stale rule from Makefile (#1782)
+  * feat: upgrade drone docker image to support multi-stage build. (#1732)
+  * Realy don't cache apk index (#1694)
+  * Limit clone depth when drone-building (#1644)
+  * Refactor Dockerfile (#1632)
+  * Check if missing/modified/unused deps in vendor and fix errors (#1468)
+  * Add GOFLAGS and EXTRA_GOFLAGS (#1438)
+  * Include formatting check to the `make test` (and thus also `check`) rule (#1366)
+* DOCS
```
Comment (tboerger) [inline CHANGELOG.md]: A docs section? seriously?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,494 @@
...
+  * Fix typos in models/ and modules/ (#1248)
+  * Refactor and fix incorrect comment (#1247)
+  * Fix migration comment (#1241)
+  * Update locale_en-US.ini (#1235)
+  * Add LibreJS support (#1201)
+  * rename OSX to macOS (#1176)
+  * add mssql to app.ini db config comment (#1172)
+  * Add MSSQL to issues template (#1171)
+* MISC
+  * Fix adding branch as protected to not allow pushing to it (#2556)
```
Comment (tboerger) [inline CHANGELOG.md]: Fix...

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,494 @@
...
+  * Fix releases to be counted from database not tags (#2389) (#2424)
```
Comment (tboerger) [inline CHANGELOG.md]: Fix...

**Review by tboerger — COMMENTED** [review]
I personally don't like this changelog, it's far too noisy and there are duplicated lines with just `gofmt` as a title.

Beside that a final release should be merged with the RCs.

File: CHANGELOG.md (hunk 5 above)
Comment (lafriks) [reply CHANGELOG.md]: It's actually only kind of fix and more like feature to be more compatible behaviour people expect from release page of repository and are used to in GitHub

Comment (lafriks) [reply CHANGELOG.md]: But it still a fix as it fixed problems with release page :)

Comment (bkcsoft) [reply CHANGELOG.md]: I'd list this as `Tags listed in releases tab (#2389) (#2424)` under Misc

Comment (lafriks) [reply CHANGELOG.md]: They were listed also in previous stable release only paging was completely broken and order also I think. They stopped showing up when paging was fixed and so I fixed that back

**Comment by lunny** [comment]
@lafriks done.

File: CHANGELOG.md (hunk 5 above)
Comment (lunny) [reply CHANGELOG.md]: @tboerger @bkcsoft done.

File: CHANGELOG.md (hunk 4 above)
Comment (lunny) [reply CHANGELOG.md]: done.

File: CHANGELOG.md (hunk 3 above)
Comment (lunny) [reply CHANGELOG.md]: Why not? Or Where should I put this changes? Misc?

File: CHANGELOG.md (hunk 2 above)
Comment (lunny) [reply CHANGELOG.md]: done.

File: CHANGELOG.md (hunk 1 above)
Comment (lunny) [reply CHANGELOG.md]: done.

**Comment by lunny** [comment]
This changelog is generated from PRs according lables on milestone 1.2.0 by github.com/go-gitea/changelog. And I have adjusted them, please review again.

**Comment by ptman** [comment]
Maybe this is a clear indication that [Using commit log diffs as changelogs is a bad idea](http://keepachangelog.com/en/1.0.0/). It can be useful to use as basis to write the changelog, so that nothing is forgotten. But clearly needs a human touch.

**Comment by MCF** [comment]
Folks, it doesn't have to be perfect for this release.  Can we please just release 1.2.0.  And thanks to everyone who has contributed, particularly the core contributors.

**Comment by lafriks** [comment]
LGTM

**Comment by daviian** [comment]
LGTM

**Pull request merged**